### PR TITLE
Fix just created folder dissapears when synchronizing parent folder

### LINKED ISSUE #2536: [BUG] Just created folder disappears when synchronizing parent folder
### Actual behaviour
Just created folder disappears when synchronizing parent folder and reappears during the next synchronization

### Expected behaviour
Just created folder should not disappear
 
### Steps to reproduce
1. Create new folder
2. Just created folder is shown successfully
3. Pull down to sync the parent folder
4. Just created folder disappears
5. Pull down again to sync the parent folder
6. Just created folder reappears

[](https://user-images.githubusercontent.com/15261363/57215733-16630b00-6fee-11e9-8a98-6fa8099117ec.gif)

### Environment data
Android version: 8.1

Device model: Nexus 5X

Stock or customized system: Stock

ownCloud app version: 2.10

ownCloud server version: 10.0.10 (stable)

### PR
App: https://github.com/owncloud/android/pull/2559

### PR DESCRIPTION
**Pull request by davigonz**
Fix #2536

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java [hunk 1]
```diff
@@ -272,30 +292,30 @@ private void removeLocalFolder() {
...
             throws OperationCancelledException {
         Log_OC.d(TAG, "Synchronizing " + mAccount.name + mRemotePath);
 
         FileDataStorageManager storageManager = getStorageManager();
 
         // parse data from remote folder
-        OCFile updatedFolder = FileStorageUtils.createOCFileFrom(
-                (RemoteFile) folderAndFiles.get(0)
+        OCFile updatedFolder = FileStorageUtils.createOCFileFromRemoteFile(
+                remoteFolderAndFiles.get(0)
```
Comment (hosy) [inline owncloudApp/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java]: Are you sure, index 0 does exists? Is there a check needed, if this element is available?

Comment (davigonz) [reply owncloudApp/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java]: `remoteFolderAndFiles` data comes from [`fetchFolderResult.getData()`](https://github.com/owncloud/android/pull/2559/files#diff-81259bfca184e3d1e468d72ab0e64ef0R226) call, whose result should contain always the root folder when [`fetchFolderResult.isSuccess()`](https://github.com/owncloud/android/pull/2559/files#diff-81259bfca184e3d1e468d72ab0e64ef0R224)

**Comment by davigonz** [comment]
@jesmrec code approved, ready to test

**Comment by jesmrec** [comment]
Checks performed:

- [X] Creation of the folder in the app
- [X] Creation of the folder in the document provider (Files/Downloads)
- [X] Rename an av. offilne folder (regression)
- [X] Move an av. offilne folder (regression)

Android 9

**Comment by jesmrec** [comment]
Do you miss something in the checks above, @davigonz? if not, approved on my side.

**Comment by davigonz** [comment]
> Do you miss something in the checks above, @davigonz? if not, approved on my side.

No, it looks good to me

**Comment by jesmrec** [comment]
Approved then

**Pull request merged**